yarn-project/prover-node/src/job/epoch-proving-job.ts

```diff
@@ -201,15 +194,14 @@ export class EpochProvingJob {
       emptyKernelOutput.end.nullifiers.map(n => n.toBuffer()),
       NULLIFIER_SUBTREE_HEIGHT,
     );
-    const allPublicDataWrites = padArrayEnd(
-      emptyKernelOutput.end.publicDataWrites.map(({ leafSlot, value }) => new PublicDataTreeLeaf(leafSlot, value)),
-      PublicDataTreeLeaf.empty(),
-      MAX_TOTAL_PUBLIC_DATA_UPDATE_REQUESTS_PER_TX,
-    );
+    const allPublicDataWrites = emptyKernelOutput.end.publicDataWrites
+      .filter(write => !write.isEmpty())
+      .map(({ leafSlot, value }) => new PublicDataTreeLeaf(leafSlot, value));
+
```

Comment:
Hmmm looks like we've been inserting empty fields to the trees for the padded/empty txs 😬 
Good that it's not for public data tree anymore now that we filter out empty writes!
We should fix it in a dedicated pr!

Reply:
Yes, a padding tx advances all the trees as any other tx currently! It's a waste but IDK if that was done to make syncing easier or something. Altough I recall seeing padding txs also in syncing so maybe it's a case where we're padding on both sides.